tests/test_location_pi.py

```diff
@@ -0,0 +1,508 @@
+                     "package_name": new_package_name,
+                     "quantity": 6}]
+
+        stock_drift = self.test_location_01._get_stock_drift(adjs_req)
+
+        self.assertEqual(len(stock_drift), 1, "Not a unique stock drift entry")
+
+        (stock_info, qty), = stock_drift.items()
+
+        self.assertEqual(qty, 6, "Wrong adjustment quantity")
+        self.assertEqual(stock_info.product_id, self.apple.id,
+                         "Wrong product id")
+        self.assertIsNotNone(stock_info.package_id,
+                             "Does not contain a package")
+
+        new_package = Package.get_package(stock_info.package_id)
+        new_package_date = datetime.strptime(new_package.create_date,
+                                             ODOO_DATETIME_FORMAT)
+
+        self.assertTrue(new_package_date >= test_start_date,
+                        "New package does not seem new")
+        self.assertFalse(stock_info.lot_id, "Wrongly added a lot id")
+
+    def test14_get_stock_drift_new_package_token_multiple_products(self):
+        """
+        Returns the correct metadata for the picking creation if
+        a request item contains a 'NEWPACKAGE' package; multiple
+        product requests, with the same package name.
+        """
+        self.create_quant(self.apple.id, self.test_location_01.id, 5)
+        self.create_quant(self.banana.id, self.test_location_01.id, 6)
+        adjs_req = [{"product_id": self.apple.id,
+                     "package_name": "tes14_NEWPACKAGE",
+                     "quantity": 15},
+                    {"product_id": self.banana.id,
+                     "package_name": "tes14_NEWPACKAGE",
+                     "quantity": 16}]
+
+        stock_drift = self.test_location_01._get_stock_drift(adjs_req)
+
+        self.assertEqual(len(stock_drift), 2,
+                         "Does not containg two stock drift entries")
+
+        package_ids = []
+
+        for stock_info, qty in stock_drift.items():
+            if stock_info.product_id == self.apple.id:
+                self.assertEqual(qty, 15, "Wrong quantity")
+            elif stock_info.product_id == self.banana.id:
+                self.assertEqual(qty, 16, "Wrong quantity")
+            else:
+                self.fail("Unexpected product id")
+
+            package_ids.append(stock_info.package_id)
+
+        self.assertEqual(package_ids[0], package_ids[1],
+                         "Did not use the same package")
+
+    def test15_get_stock_drift_no_package_token_with_lot(self):
+        """
+        Returns the correct metadata for the picking creation if
+        a request item contains a 'NO_PACKAGE' package and a lot
+        is specified - the lot should be created if the product
+        is tracked.
+        """
+        Lot = self.env['stock.production.lot']
+
+        lot_name = "test_15_lot"
+        self.apple.tracking = 'serial'
+        self.create_quant(self.apple.id, self.test_location_01.id, 7)
```

Comment:
A quant for a serial tracked item should have the lot_id also set and qty = 1.

Reply:
will create another test, with an existent lot for that